files/KoreBuild/scripts/commands/Invoke-MSBuild.ps1

```diff
@@ -0,0 +1,137 @@
+    }
+
+    $Path = $global:Path
+    Push-Location $Path | Out-Null
+    try {
+        Write-Verbose "Building $Path"
+        Write-Verbose "dotnet = ${global:dotnet}"
+
+        # Generate global.json to ensure the repo uses the right SDK version
+        $sdkVersion = $global:SDKVersion
+        if ($sdkVersion -ne 'latest') {
+            "{ `"sdk`": { `"version`": `"$sdkVersion`" } }" | Out-File (Join-Path $Path 'global.json') -Encoding ascii
+        } else {
+            Write-Verbose "Skipping global.json generation because the `$sdkVersion = $sdkVersion"
+        }
+
+        $makeFileProj = Join-Paths $PSScriptRoot ('../..', 'KoreBuild.proj')
+        $msbuildArtifactsDir = Join-Paths $Path ('artifacts', 'msbuild')
+        $msBuildResponseFile = Join-Path $msbuildArtifactsDir msbuild.rsp
+
+        $msBuildLogArgument = ""
+
+        if ($VerbosePreference -eq 'Continue' -or $env:KOREBUILD_ENABLE_BINARY_LOG -eq "1") {
+            Write-Verbose 'Enabling binary logging'
+            $msbuildLogFilePath = Join-Path $msbuildArtifactsDir msbuild.binlog
+            $msBuildLogArgument = "/bl:$msbuildLogFilePath"
+        }
+
+        $msBuildArguments = @"
+/nologo
+/m
+/p:RepositoryRoot="$Path/"
+"$msBuildLogArgument"
+/clp:Summary
+"$makeFileProj"
+"@
+
+        $MSBuildArgs | ForEach-Object { $msBuildArguments += "`n`"$_`"" }
+
+        if (!(Test-Path $msbuildArtifactsDir)) {
+            New-Item -Type Directory $msbuildArtifactsDir | Out-Null
+        }
+
+        $msBuildArguments | Out-File -Encoding ASCII -FilePath $msBuildResponseFile
+
+        $noop = ($MSBuildArgs -contains '/t:Noop' -or $MSBuildArgs -contains '/t:Cow')
+        Write-Verbose "Noop = $noop"
+        $firstTime = $env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE
+        if ($noop) {
+            $env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE = 'true'
+        }
+        else {
+            __build_task_project $Path
+        }
+
+        Write-Verbose "Invoking msbuild with '$(Get-Content $msBuildResponseFile)'"
+
+        __exec $global:dotnet msbuild `@"$msBuildResponseFile"
+    }
+    catch{
+        Write-Host "Error: $_"
+    }
+    finally {
+        Pop-Location
+        $env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE = $firstTime
+    }
+}
+
+#
+# Helpers
+#
+function __build_task_project($RepoPath) {
+    $taskProj = Join-Paths $RepoPath ('build', 'tasks', 'RepoTasks.csproj')
+    $publishFolder = Join-Paths $RepoPath ('build', 'tasks', 'bin', 'publish')
+
+    if (!(Test-Path $taskProj)) {
+        return
+    }
+
+    if (Test-Path $publishFolder) {
+        Remove-Item $publishFolder -Recurse -Force
+    }
+
+    $sdkPath = "/p:RepoTasksSdkPath=$(Join-Paths $PSScriptRoot ('..', 'msbuild', 'KoreBuild.RepoTasks.Sdk', 'Sdk'))"
+
+    __exec $global:dotnet restore $taskProj $sdkPath
+    __exec $global:dotnet publish $taskProj --configuration Release --output $publishFolder /nologo $sdkPath
+}
+
+function __exec($cmd) {
```

Comment:
We will likely need this in other places. It might be good to create a `common.psm1` file that individual commands can import.
```
- scripts/
    + common.psm1
    + KoreBuild.psm1
- commands/
     + Example.psm1
```
```ps1
# commands/Example.psm1

Import-Module "$PSScriptRoot/../common.psm1"
```